internal/templater/funcs.go

```diff
@@ -52,6 +52,16 @@ func init() {
 		"relPath": func(basePath, targetPath string) (string, error) {
 			return filepath.Rel(basePath, targetPath)
 		},
+		"merge": func(a, b map[string]any) map[string]any {
+			m := make(map[string]any)
```

Comment:
Idea: set the initial capacity to the length of the existing maps:

```suggestion
			m := make(map[string]any, len(a)+len(b))
```

Reply:
I did think about this but didn't add it since if there are duplicate keys in both maps, we will be creating up to 2x more capacity than needed. However, I guess this isn't a huge problem as we're probably optimising for allocations rather than memory efficiency.

I've added this change and also included the new function in the docs.

Reply:
I realized now that we could have made this function variadic, accepting any number of maps instead of just 2. Good that it can still be made without being backwards incompatible.

Reply:
That's a great idea. See #1464